### ISSUE TITLE
Missing credentials error - welcome message with 30s delay/unresponsiveness

### ISSUE THREAD

**Issue by brack** [awslabs/amazon-qldb-shell#182]
Backlog item for known issue in v2, such that if aws credentials are not configured (or potentially in other scenarios such as not having proper IAM access configured), the tool will become unresponsive for ~30 seconds after showing a welcome message.

Once AWS Rust SDK is updated, we can update and address this delay

**Comment by marcbowes** [comment]
The root cause is from the rusoto credential provider chain, which includes the EC2 instance metadata provider. It can time out in HTTP, probably due to firewall rules or something.

**Comment by marcbowes** [comment]
https://github.com/awslabs/smithy-rs/pull/709/files implements the IMDS creds in the new SDK and will have a 1s timeout instead of a 30s timeout. I think that's pretty reasonable, as running without creds is an error condition.

**Comment by marcbowes** [comment]
The current default creds chain comes from rusoto and is

```
#[derive(Debug, Clone)]
pub struct ChainProvider {
    environment_provider: EnvironmentProvider,
    instance_metadata_provider: InstanceMetadataProvider,
    container_provider: ContainerProvider,
    profile_provider: Option<ProfileProvider>,
}
```

with the above referenced PR, the sdk chain is now:

```
            let provider_chain = CredentialsProviderChain::first_try("Environment", env_provider)
                .or_else("Profile", profile_provider)
                .or_else("WebIdentityToken", web_identity_token_provider)
                .or_else("Ec2InstanceMetadata", imds_provider);
```

which is pretty close, missing only the "container provider". I think we should flip over, as I don't believe anybody is using the shell within a container.

**Comment by marcbowes** [comment]
Fixed via #208
